common/src/main/java/io/homeassistant/companion/android/common/sensors/SensorReceiverBase.kt

```diff
@@ -0,0 +1,177 @@
+package io.homeassistant.companion.android.common.sensors
+
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.content.res.Configuration
+import android.util.Log
+import io.homeassistant.companion.android.common.data.integration.IntegrationRepository
+import io.homeassistant.companion.android.common.data.integration.SensorRegistration
+import io.homeassistant.companion.android.database.AppDatabase
+import kotlinx.coroutines.*
+import java.util.*
+import javax.inject.Inject
+
+abstract class SensorReceiverBase : BroadcastReceiver() {
+    private val ioScope: CoroutineScope = CoroutineScope(Dispatchers.IO + Job())
+
+    protected abstract val tag: String
+    protected abstract val currentAppVersion: String
+    protected abstract val managers: List<SensorManager>
+
+    @Inject
+    lateinit var integrationUseCase: IntegrationRepository
```

Comment:
So I tested the latest APK and I did not see any sensors get created. Then I took my watch off the charger and got a crash that led to this line. I believe we may need to add `@AndroidEntryPoint` above line 15 because we are injecting here.  Not 100% on that though.

```
2021-11-26 00:50:49.670 18559-18665/io.homeassistant.companion.android.debug E/AndroidRuntime: FATAL EXCEPTION: DefaultDispatcher-worker-1
    Process: io.homeassistant.companion.android.debug, PID: 18559
    kotlin.UninitializedPropertyAccessException: lateinit property integrationUseCase has not been initialized
        at io.homeassistant.companion.android.common.sensors.SensorReceiverBase.getIntegrationUseCase(SensorReceiverBase.kt:23)
        at io.homeassistant.companion.android.common.sensors.SensorReceiverBase$onReceive$2.invokeSuspend(SensorReceiverBase.kt:80)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)
```

Strange that this error occurs when our registered intent is fired but the worker itself says its successful without registering any sensors.

```
2021-11-26 00:52:22.968 19057-19087/io.homeassistant.companion.android.debug I/WM-WorkerWrapper: Worker result SUCCESS for Work [ id=03ca93f0-c6bd-4bbb-b678-0410d67ebc77, tags={ io.homeassistant.companion.android.sensors.SensorWorker } ]
```

Reply:
Hmm, so it looks that somethings changed that I haven't incorporated in the wear app in the last 2 weeks. I think I've now got a working imlementation which was taken from the mobile app as part of the check for disabled location. Calling `SensorManager#isEnabled()` seems to be the only way to add the sensors to the DB.

https://github.com/home-assistant/android/pull/1890/commits/0d6660952311433973f2aa3f9a443c47a6c291b3#diff-e4be6176c8e8184dc31ba7addb801b46028bbaa439cdd48b80183296dc96d6f3R59

Reply:
The biggest impactful update I can see is either the Entity Category PR or the Hilt upgrade PR as both touched sensors and how we register them.  I am not sure about why calling `SensorManager.isEnabled()` is the only way to get them registered because we have `enabledByDefault` to keep track of what should be enabled upon onboarding.

Reply:
What I mean is that to acutally add the sensors to the database, it seems that the only way is through the `SensorManager#isEnabled()` call. Although they are enabled by default, the sensors aren't added to the DB so there's nothing to check! The worker conks out becuase there are no sensors in the database, thus no enabled sensors so it's all like, there's nothing to do here. Do you get me?

https://github.com/Kisty/home-assistant-android/blob/0d6660952311433973f2aa3f9a443c47a6c291b3/common/src/main/java/io/homeassistant/companion/android/common/sensors/SensorManager.kt#L60-L64

https://github.com/Kisty/home-assistant-android/blob/0d6660952311433973f2aa3f9a443c47a6c291b3/common/src/main/java/io/homeassistant/companion/android/common/sensors/SensorWorkerBase.kt#L37

Reply:
Ah yes you are right about that. I thought we used to do this in SensorReceiver but I think we moved it and I had forgotten about that, oops LOL. The only places we use `sensorDao.add(sensorId)` is in the location handler, onboarding for location sensors and  in the sensor fragments as a sensor gets enabled.  Good catch!  Looking forward to testing once the actions run :)

Reply:
OK, cool. Are you able to approve the actions?

Would it be worth (maybe in another PR) to add the sensors somewhere else explicitly or leave it as-is?

Reply:
>  Are you able to approve the actions?

Im not able to :( otherwise I would've approved them :P

> Would it be worth (maybe in another PR) to add the sensors somewhere else explicitly or leave it as-is?

umm lets see what comes of the review to see if it should be moved. For now i think its ok since users will come to the home screen after logging in.  Later on when we add any sensor that requires location permissions we probably want to start using the location handler anyways

Reply:
ok the actions ran and after logging in I am able to see the sensors in HA and the intents are working!

![image](https://user-images.githubusercontent.com/1634145/143663736-2c15127d-868a-4006-8985-af13403ccd78.png)

![image](https://user-images.githubusercontent.com/1634145/143663737-ae8164f7-f73d-4269-bc78-cb8b1d62c157.png)

Reply:
That's fantastic! It's a long awaited feature. For it to work for someone else is great! Going to be a big help for a bunch of users!